### PR TITLE
Add python.imread & python.mahotas

### PR DESCRIPTION
**Pull request by luispedro**
These are two packages I developed and now myself and  many other regularly use.

### CODE REVIEW

**Comment by FRidh** [comment]
Hi @luispedro, thanks for your contribution. Nice to see other scientists here adding Python packages.
Could you update the commit messages according to the nixpkgs guide.
http://nixos.org/nixpkgs/manual/#chap-submitting-changes

File: pkgs/top-level/python-packages.nix [hunk 1]
```diff
@@ -3097,6 +3097,35 @@ in modules // {
...
+      sha256 = "30c4b979e0d5f4c013860321766a79ffcabe56c1ad9088e5d0c6b36aec5f0415";
+    };
+
+    buildInputs = with self; [
+            nose
+            pillow
+            scipy
+          ];
+    propagatedBuildInputs = with self; [
+            python
```
Comment (FRidh) [inline pkgs/top-level/python-packages.nix]: no need to add python here

File: pkgs/top-level/python-packages.nix [hunk 2]
```diff
@@ -4940,6 +4969,36 @@ in modules // {
...
+    src = pkgs.fetchurl {
+      url = "https://github.com/luispedro/imread/archive/release-${version}.tar.gz";
+      sha256 = "12d7ba3523ba50d67d526e9797e041021dd9cd4acf9567a9bf73c8ae0b689d4a";
+    };
+
+    buildInputs = with self; [
+            nose
+          ];
+    propagatedBuildInputs = with self; [
+            python
```
Comment (FRidh) [inline pkgs/top-level/python-packages.nix]: nor here

File: pkgs/top-level/python-packages.nix [hunk 3]
```diff
@@ -3097,6 +3097,35 @@ in modules // {
...
+            numpy
+            imread
+          ];
+
+    meta = with stdenv.lib; {
+      description = "Computer vision package based on numpy";
+      homepage = https://readthedocs.org/projects/mahotas/;
+      maintainers = with maintainers; [ luispedro ];
+      license = licenses.mit;
+      platforms = platforms.linux;
```
Comment (FRidh) [inline pkgs/top-level/python-packages.nix]: Is darwin/osx intentionally not supported?

**Comment by luispedro** [comment]
Thanks. This is my first attempt at packaging things and I'm still getting the hang of it. I updated the PR according to your comments.

Cheers, Luis

**Comment by FRidh** [comment]
cc @domenkozar

File: pkgs/top-level/python-packages.nix [hunk 4]
```diff
@@ -3097,6 +3097,34 @@ in modules // {
...
+
+    buildInputs = with self; [
+            nose
+            pillow
+            scipy
+          ];
+    propagatedBuildInputs = with self; [
+            numpy
+            imread
+          ];
```
Comment (domenkozar) [inline pkgs/top-level/python-packages.nix]: (nitpick) please indent with two spaces

File: pkgs/top-level/python-packages.nix [hunk 5]
```diff
@@ -3097,6 +3097,34 @@ in modules // {
...
+    version = "1.4.0";
+
+    src = pkgs.fetchurl {
+      url = "https://github.com/luispedro/mahotas/archive/release-${version}.tar.gz";
+      sha256 = "30c4b979e0d5f4c013860321766a79ffcabe56c1ad9088e5d0c6b36aec5f0415";
+    };
+
+    buildInputs = with self; [
+            nose
+            pillow
```
Comment (domenkozar) [inline pkgs/top-level/python-packages.nix]: Is `pillow` not a runtime dependency?

File: pkgs/top-level/python-packages.nix [hunk 6]
```diff
@@ -4940,6 +4968,35 @@ in modules // {
...
+
+    buildInputs = with self; [
+            nose
+          ];
+    propagatedBuildInputs = with self; [
+            numpy
+            pkgs.libjpeg
+            pkgs.libpng
+            pkgs.libtiff
+            pkgs.libwebp
```
Comment (domenkozar) [inline pkgs/top-level/python-packages.nix]: these packages shouldn't be propagated, they're just `buildInputs`

File: pkgs/top-level/python-packages.nix (hunk 5 above)
Comment (luispedro) [inline pkgs/top-level/python-packages.nix]: No, just used in the test suite.

**Comment by luispedro** [comment]
Accidentally closed and github won't let me reopen.

**Pull request closed without merging**